var/spack/repos/builtin/packages/rivet/package.py

```diff
@@ -72,9 +73,17 @@ class Rivet(AutotoolsPackage):
 
     depends_on("hepmc", when="hepmc=2")
     depends_on("hepmc3", when="hepmc=3")
+    # The fix for working around patch-level zero issues have landed in 4.0.1
+    # and have also been back-ported to 3.1.11. Hence we need to exclude 4.0.0
+    # explicitly and catch the rest with versions up to 3.1.10
+    # See: https://gitlab.com/hepcedar/rivet/-/merge_requests/904
+    # and: https://gitlab.com/hepcedar/rivet/-/merge_requests/912
     conflicts(
-        "hepmc@3.3.0", when="@:4.0.0 hepmc=3", msg="patch-level zero requires at least 4.0.1"
+        "hepmc@3.3.0", when="@:3.1.10 hepmc=3", msg="patch-level zero requires at least 3.1.11"
```

Comment:
```suggestion
        "hepmc@3.3.0", when="@:3.1.10,4.0.0 hepmc=3", msg="patch-level zero requires at least 3.1.11 or 4.0.1"
```

Reply:
(and rm next diff line)

Reply:
Done. I have also moved the `hepmc=2` conflict into the `conditional` value statement of the variant.